src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -241,6 +398,241 @@ else if (!period.isEmpty()) {
+                node = child;
+            }
+            // If nested inside html tag
+            else if (child.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                node = getInlineTagNodeWithinHtmlElement(child);
+            }
+
+            if (node != null) {
+                break;
+            }
+        }
+        return node;
+    }
+
+    /**
+     * Returns inline javadoc tag node inside html tags from a HTML tag.
+     *
+     * @param ast html tag node.
+     * @return Inline summary javadoc tag node or null if no node is found.
+     */
+    private static DetailNode getInlineTagNodeWithinHtmlElement(DetailNode ast) {
+        DetailNode node = ast;
+        DetailNode result = null;
+        // node can never be null as it is always called when there is HTML element
+        if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+            result = node;
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_TAG) {
+            // HTML_TAG always has more than 2 children.
+            node = node.getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT
+                // Condition for SINGLETON html element which cannot contain summary node
+                && node.getChildren()[0].getChildren().length > 1) {
+            // Html elements have one tested tag before actual content inside it
+            node = node.getChildren()[0].getChildren()[1];
+            result = getInlineTagNodeWithinHtmlElement(node);
+        }
+        return result;
+    }
+
+    /**
+     * Checks if the first tag inside ast is summary tag.
+     *
+     * @param javadoc root node.
+     * @return true, if first tag is summary tag.
+     */
+    private static boolean isSummaryTag(DetailNode javadoc) {
+        final DetailNode[] child = javadoc.getChildren();
+
+        // Checking size of ast is not required, since ast contains
+        // children of Inline Tag, as at least 2 children will be present which are
+        // RCURLY and LCURLY.
+        return child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                && SUMMARY_TEXT.equals(child[1].getText());
+    }
+
+    /**
+     * Check the summary of inline form.
+     *
+     * @param ast Javadoc root Node.
+     */
+    private void validateSummaryTag(DetailNode ast) {
+        final String inlineSummary = getInlineSummary(ast);
+        final String summaryVisible = getVisibleContent(inlineSummary);
+        if (summaryVisible.isEmpty()) {
+            log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC_MISSING);
+        }
+        else if (!period.isEmpty()) {
+            if (isPeriodAtEnd(summaryVisible, period)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_MISSING_PERIOD);
+            }
+            else if (!containsCorrectInlineTags(ast)) {
+                log(ast.getLineNumber(), MSG_SUMMARY_JAVADOC);
```

Comment:
`if (!containsCorrectInlineTags)`

is nonsense, I do not undestand why Check care about inlinetags, at this place it should care about `containsForbiddenFragment`

Reply:
> why Check care about inlinetags
Because all tags are not permitted inside javadoc

> at this place it should care about containsForbiddenFragment
Added this function.

Reply:
> Because all tags are not permitted inside javadoc

ok, I found a confusion point ....
this validation will be done by javadoc tool. It is not a goal of this Check, but I agree that name of Check is very generic but description of Check narrow target of validation to certain items only.
Checkstyle suppose to work only on code that passing on javadoc tool execution. The same as do for java code, we do not support execution of checkstyle on non-compiled code.

-----

we follow unix model of design for Checks - there should be no Checks that do all validation, each Check validate certain item, user need to use combination of Checks to get required result.

Reply:
Done, removed check of inline tags.